compound.hh

```diff
@@ -187,8 +186,7 @@ public:
             ++(*this);
             return i;
         }
-        const value_type& operator*() const { return _current; }
-        const value_type* operator->() const { return &_current; }
```

Comment:
ּRemoving this operator forced you to make dozens of ugly changes to the code below. It's sad that it's necessary... The discussion about "stable pointers" doesn't apply to any of the uses of this operator-> that we had in the code, as far as I can tell.

Reply:
It's not good to promise something you don't keep.